src/btree/bt_vrfy.c

```diff
@@ -582,6 +583,115 @@ __verify_addr_ts(WT_SESSION_IMPL *session, WT_REF *ref, WT_CELL_UNPACK *unpack,
     return (0);
 }
 
+/*
+ * __verify_btree_id_with_meta --
+ *     With a given btree id, iterate through the metafile table and check if the btree id exists in
+ *     any table. If exists, we place the table URI into a variable Otherwise should return
+ *     WT_NOTFOUND.
+ */
+static int
+__verify_btree_id_with_meta(WT_SESSION_IMPL *session, uint32_t btree_id, const char **uri)
+{
+    WT_CONFIG_ITEM id;
+    WT_CURSOR *meta_cursor;
+    WT_DECL_RET;
+    const char *meta_value;
+
+    WT_ERR(__wt_metadata_cursor(session, &meta_cursor));
+
+    WT_ERR(meta_cursor->reset(meta_cursor));
+    while ((ret = meta_cursor->next(meta_cursor)) == 0) {
+        meta_cursor->get_value(meta_cursor, &meta_value);
+        WT_ERR_NOTFOUND_OK(__wt_config_getones(session, meta_value, "id", &id));
+        if (ret == WT_NOTFOUND) {
+            continue;
+        }
+        if (errno != ERANGE && btree_id == __wt_strtouq(id.str, NULL, 10)) {
+            meta_cursor->get_key(meta_cursor, uri);
+            goto done;
+        }
+    }
+
+done:
+err:
+    WT_TRET(__wt_metadata_cursor_release(session, &meta_cursor));
+    return (ret);
+}
+
+/*
+ * __wt_verify_history_store_tree --
+ *     Verify the history store tree, by iterating and checking all the keys in the history and that
+ *     it exists in both the meta file and the respective data store table. Skips duplicate keys by
+ *     storing the previous key.
```

Comment:
Could we strengthen this comment by talking about exactly what relationships we're verifying here? I've read the comment and the ticket, and I still don't understand exactly what we're trying to "verify".

The comment "Update wt verify to check that a key in history store is missing from the data store." in the ticket is particularly confusing.